lib/shopify_app/controller_concerns/login_protection.rb

```diff
@@ -17,6 +17,11 @@ class ShopifyDomainNotFound < StandardError; end
     ACCESS_TOKEN_REQUIRED_HEADER = 'X-Shopify-API-Request-Failure-Unauthorized'
 
     def activate_shopify_session
+      if user_session_expected? && user_session.blank?
+        signal_access_token_required
+        return redirect_to_login
```

Comment:
`redirect_to_login` expects the request to have the header `HTTP_X_REQUESTED_WITH: XMLHttpRequest` to indicate it's an xhr request.

This header is not set by out current utility fetch methods, nor is it set by apollo client when interacting with its graphql client.

To fix this, we can either:
1. Add that header as part of the `authenticatedFetch` definition or
2. Change `if request.xhr?` to something that encompasses a different type of request altogether

Reply:
I think we have to set the header because otherwise there's no way for the backend to differentiate between AJAX and regular requests.

Reply:
Should this be the default behaviour in App Bridge's `authenticatedFetch` method?

cc: @vividviolet

Reply:
I'm fine with putting it in authenticatedFetch. We only want authenticatedFetch to be used for xhr requests anyway

Reply:
@NabeelAhsen can you link to the app-bridge branch you have for adding the header for reference?

Reply:
Makes sense to add it to App Bridge but do we not need a fallback in the case where they're not using any of the App Bridge utils?

Reply:
Here's the related PR: https://github.com/Shopify/app-bridge/pull/1652

My assumption, at least with the flow described here, is that the fallback case is: if they're using `xhr`/ajax requests, they're still responsible for handling any unauthorized responses that Shopify App returns.